File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/repository/support/RequeryRepositoryImplementation.java
package com.coupang.springframework.data.requery.repository.support;

import com.coupang.springframework.data.requery.repository.RequeryPredicatedExecutor;
import com.coupang.springframework.data.requery.repository.RequeryRepository;

/**
 * com.coupang.springframework.data.requery.repository.support.RequeryRepositoryImplementation
 *
 * @author debop
 * @since 18. 6. 6
 */
public interface RequeryRepositoryImplementation<T, ID> extends RequeryRepository<T, ID>, RequeryPredicatedExecutor<T> {

    void setRepositoryMethodMetadata(CrudMethodMetadata crudMethodMetadata);
}

File: src/main/kotlin/com/coupang/springframework/data/requery/repository/support/RequeryRepositoryImplementation.kt
package com.coupang.springframework.data.requery.repository.support

import com.coupang.springframework.data.requery.repository.RequeryRepository
import com.coupang.springframework.data.requery.repository.RequerySelectionExecutor
import org.springframework.data.repository.NoRepositoryBean

/**
 * RequeryRepositoryImplementation
 *
 * @author debop@coupang.com
 * @since 18. 5. 29
 */
@NoRepositoryBean
interface RequeryRepositoryImplementation<T, ID>: RequeryRepository<T, ID>, RequerySelectionExecutor<T> {

    // JPA에만 있다
    // fun setRepositoryMethodMetadata(crudMethodMetadata:CrudMethodMetadata)

}